projects/power_distribution/test/test_fan_ctrl.c

```diff
@@ -0,0 +1,219 @@
+#include <stdbool.h>
+#include "adc.h"
+#include "adt7476a_fan_controller.h"
+#include "adt7476a_fan_controller_defs.h"
+#include "can.h"
+#include "can_transmit.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "i2c.h"
+#include "interrupt.h"
+#include "log.h"
+#include "ms_test_helpers.h"
+#include "pd_fan_ctrl.h"
+#include "pin_defs.h"
+#include "status.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+#define TEST_I2C_PORT I2C_PORT_2
+
+#define TEST_CONFIG_PIN_I2C_SCL \
+  { GPIO_PORT_B, 10 }
+#define TEST_CONFIG_PIN_I2C_SDA \
+  { GPIO_PORT_B, 11 }
+
+#define TEST_PWM_1 ADT_PWM_PORT_1
+#define TEST_PWM_2 ADT_PWM_PORT_2
+
+#define ADT7476A_PWM_1 0x30
+#define ADT7476A_PWM_3 0x32
+
+#define OVERTEMP_FLAGS (FAN_OVERTEMP_TRIGGERED | DCDC_OVERTEMP | ENC_VENT_OVERTEMP)
+#define OVERVOLT_FLAGS (VCC_EXCEEDED | VCCP_EXCEEDED)
+#define FAN_ERR_FLAGS (FAN1_STATUS | FAN2_STATUS | FAN3_STATUS | FAN4_STATUS)
+
+#define FAN_UNDERTEMP_VOLTAGE 1163
+#define FAN_OVERTEMP_VOLTAGE 758
+#define FAN_50_PERC_VOLTAGE 975
+
+#define FAN_MAX_I2C_WRITE 0xFF  // equates to value/0.39 -> adt7476A conversion
+#define FAN_50_PERC_I2C_WRITE 0x7F
+#define FAN_MIN_I2C_WRITE 0x0
+
+#define ADC_MAX_VAL 3300
+
+typedef enum {
+  TEST_CAN_EVENT_RX = 10,
+  TEST_CAN_EVENT_TX,
+  TEST_CAN_EVENT_FAULT,
+  NUM_TEST_CAN_EVENTS,
+} TestCanEvent;
```

Comment:
You could just use the power distribution CAN events from `pd_events.h`, that way you know they won't conflict